pallets/xyk-liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,927 @@
+				who,
+				amount: shares_amount,
+				lp_token: amm_share_token,
+				deposit_id,
+			});
+
+			Ok(())
+		}
+
+		/// This function create yield farm entry for existing deposit. LP shares are not transferred
+		/// and amount of LP shares is based on existing deposit.
+		///
+		/// This function DOESN'T create new deposit.
+		///
+		/// Parameters:
+		/// - `origin`: account depositing LP shares. This account have to have at least
+		/// - `global_farm_id`: global farm identifier.
+		/// - `yield_farm_id`: yield farm identifier redepositing to.
+		/// - `asset_pair`: asset pair identifying LP shares user want to deposit.
+		/// - `deposit_id`: identifier of the AMM pool.
+		///
+		/// Emits `SharesRedeposited` event when successful.
+		#[pallet::weight(<T as Config>::WeightInfo::redeposit_lp_shares())]
+		pub fn redeposit_lp_shares(
+			origin: OriginFor<T>,
+			global_farm_id: GlobalFarmId,
+			yield_farm_id: YieldFarmId,
+			asset_pair: AssetPair,
+			deposit_id: DepositId,
+		) -> DispatchResult {
+			let owner = Self::ensure_nft_owner(origin, deposit_id)?;
+			Self::ensure_amm(asset_pair)?;
+
+			let amm_share_token = T::AMM::get_share_token(asset_pair);
+
+			let shares_amount = T::LiquidityMiningHandler::redeposit_lp_shares(
+				global_farm_id,
+				yield_farm_id,
+				deposit_id,
+				Self::get_asset_balance_in_amm_pool,
+			)?;
+
+			Self::deposit_event(Event::SharesRedeposited {
+				global_farm_id,
+				yield_farm_id,
+				who: owner,
+				amount: shares_amount,
+				lp_token: amm_share_token,
+				deposit_id,
+			});
+
+			Ok(())
+		}
+
+		/// Claim rewards from liq. mining for deposit represented by `nft_id`.
+		///
+		/// This function calculate user rewards from liq. mining and transfer rewards to `origin`
+		/// account. Claiming in the same period is allowed only once.
+		///
+		/// Parameters:
+		/// - `origin`: account owner of deposit(nft).
+		/// - `deposit_id`: nft id representing deposit in the yield farm.
+		/// - `yield_farm_id`: yield farm identifier to claim rewards from.
+		///
+		/// Emits `RewardClaimed` event when successful.
+		#[pallet::weight(<T as Config>::WeightInfo::claim_rewards())]
+		pub fn claim_rewards(
+			origin: OriginFor<T>,
+			deposit_id: DepositId,
+			yield_farm_id: YieldFarmId,
+		) -> DispatchResult {
+			let owner = Self::ensure_nft_owner(origin, deposit_id)?;
+
+			let fail_on_double_claim = true;
+			let (global_farm_id, reward_currency, claimed, _) = T::LiquidityMiningHandler::claim_rewards(
+				owner.clone(),
+				deposit_id,
+				yield_farm_id,
+				fail_on_double_claim,
+			)?;
+
+			ensure!(!claimed.is_zero(), Error::<T>::ZeroClaimedRewards);
+
+			Self::deposit_event(Event::RewardClaimed {
+				global_farm_id,
+				yield_farm_id,
+				who: owner,
+				claimed,
+				reward_currency,
+				deposit_id,
+			});
+
+			Ok(())
+		}
+
+		/// Withdraw LP shares from liq. mining. with reward claiming if possible.
+		///
+		/// Cases for transfer LP shares and claimed rewards:
```

Comment:
this sentence doesn't make much sense to me. 

List of possible cases of transfers of shares and rewards ?